This one is committed upstream as r819560 - and can thus be deleted when we get there


Index: b/kdecore/kernel/kstandarddirs.cpp
===================================================================
--- a/kdecore/kernel/kstandarddirs.cpp
+++ b/kdecore/kernel/kstandarddirs.cpp
@@ -957,6 +957,8 @@
 
         QStringList dirs;
         dirs = d->relatives.value(type);
+        QString installdir = installPath( type );
+        QString installprefix = installPath("kdedir");
 
         if (!dirs.isEmpty())
         {
@@ -995,24 +997,33 @@
                  pit != prefixList->end();
                  ++pit)
             {
-                for (QStringList::ConstIterator it = dirs.begin();
-                     it != dirs.end(); ++it)
-                {
-                    if ( (*it).startsWith('%'))
-                        continue;
-                    QString path = realPath( *pit + *it );
-                    testdir.setPath(path);
-                    if (local && restrictionActive)
-                        continue;
-                    if ((local || testdir.exists()) && !candidates.contains(path))
-                        candidates.append(path);
+	        if((*pit)!=installprefix||installdir.isEmpty())
+	        {
+                    for (QStringList::ConstIterator it = dirs.begin();
+                         it != dirs.end(); ++it)
+                    {
+                        if ( (*it).startsWith('%'))
+                            continue;
+                        QString path = realPath( *pit + *it );
+                        testdir.setPath(path);
+                        if (local && restrictionActive)
+                            continue;
+                        if ((local || testdir.exists()) && !candidates.contains(path))
+                            candidates.append(path);
+                    }
+                    local = false;
                 }
-                local = false;
-            }
+	        else
+	        {
+                    // we have a custom install path, so use this instead of <installprefix>/<relative dir>
+	            testdir.setPath(installdir);
+                    if(testdir.exists() && ! candidates.contains(installdir))
+                        candidates.append(installdir);
+	        }
+	    }
         }
 
         // make sure we find the path where it's installed
-        QString installdir = installPath( type );
         if (!installdir.isEmpty()) {
             bool ok = true;
             foreach (const QString &s, candidates) {
